src/Core/src/Platform/iOS/MauiUIApplicationDelegate.cs

```diff
@@ -1,53 +1,117 @@
 using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Linq;
 using Foundation;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
-using UIKit;
 using Microsoft.Maui.Hosting;
+using UIKit;
 
 namespace Microsoft.Maui
 {
-	public class MauiUIApplicationDelegate<TApplication> : UIApplicationDelegate, IUIApplicationDelegate where TApplication : MauiApp
+	public class MauiUIApplicationDelegate<TStartup, TApplication> : UIApplicationDelegate, IUIApplicationDelegate
+		where TStartup : IStartup
+		where TApplication : Application
 	{
+		bool _isSuspended;
+		Application? _app;
+		IWindow? _window;
+
 		public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
 		{
+			if (!(Activator.CreateInstance(typeof(TStartup)) is TStartup startup))
+				throw new InvalidOperationException($"We weren't able to create the Startup {typeof(TStartup)}");
+
+			var appBuilder = AppHostBuilder
+				.CreateDefaultAppBuilder()
+				.ConfigureServices(ConfigureNativeServices);
+
+			startup.Configure(appBuilder);
+
+			appBuilder.Build();
+
 			if (!(Activator.CreateInstance(typeof(TApplication)) is TApplication app))
```

Comment:
I think part of the idea of having the startup class is so that the app is resolved via services so that they an be used in the app's ctor. It feels like the builder should create the host and services collection, and then we query for the `IApplication`. Then people can have ctor args of `IServiceProvider` or `IMauiContext` and use it in the ctor itself.

Reply:
This is also going to go a bit extreme, but we probably don't need to have a static `Application.Current` because if the app is in the services, then if anyone wants the app, all they have to do is add a ctor arg and they will magically get it. And, if all our things are resolved via services, anyone can get it that way - handlers, view and whatever was registered. Having a static `Current` breaks the DI system a bit, especially when testing - this is the reason for the collections in the first place. If someone makes an app that they want to track the current, all they need is to set the property on their type.

Reply:
After going for a nap I think this might be too extreme for the non-DI folks... If you want pure DI, then don't use the statics, if not, it is there for you.